src/components/cards/MetadataCard.jsx

```diff
@@ -34,38 +64,57 @@ export default function MetadataCard({
       }
     >
       <List dense>
-        {metadata.map(datum => {
-          const value = datum.value || datum.defaultValue;
-          if (!value) return null;
-          return (
-            <ListItem key={datum.id}>
-              {datum.icon && (
-                <ListItemIcon style={{ minWidth: 36 }}>
-                  <datum.icon />
-                </ListItemIcon>
-              )}
-              <ListItemText
-                primary={
-                  <Text
-                    component="span"
-                    variant="caption"
-                    id={datum.titleId}
+        {metadata
+          ? metadataToDisplay.map(field => {
+              const viewComponentProps =
+                field.viewComponentProps || {};
+
+              return (
+                <ListItem key={field.id || field.name}>
+                  {field.icon && (
+                    <ListItemIcon style={{ minWidth: 36 }}>
+                      <field.icon />
+                    </ListItemIcon>
+                  )}
+                  <ListItemText
+                    primary={
+                      <Text
+                        component="span"
+                        variant="caption"
+                        id={field.labelId}
+                      >
+                        {field.label}
+                      </Text>
+                    }
+                    secondary={
+                      <field.viewComponent
+                        value={field.value}
+                        choices={field.choices}
+                        schema={field}
+                        {...viewComponentProps}
+                      />
+                    }
                   />
-                }
-                secondary={
-                  datum.renderValue ? (
-                    datum.renderValue(value)
-                  ) : (
-                    <Text component="span" variant="body2">
-                      {value}
-                    </Text>
-                  )
-                }
-              />
-            </ListItem>
-          );
-        })}
+                </ListItem>
+              );
+            })
+          : range(6).map(i => (
```

Comment:
Why 6?

Reply:
Oh. Only if it's completely empty, right? Then populate with six skeleton items?

Reply:
yes